start-local-vm: allow resizing the extracted images before boot

markusboehme opened the pull request:
**Issue number:** n/a

**Description of changes:** Sometimes it is helpful to not go with the default image sizes for testing purposes--in practice, this would mean testing with bigger images. Introduce the `--boot-image-size` and `--data-image-size` options to `start-local-vm` that allow for resizing the respective images after they have been extracted, but before they are booted.

**Testing done:**

Without `--boot-image-size`, the regular metal-dev disk image size is used:

```
bash-5.1# df -h /local
Filesystem      Size  Used Avail Use% Mounted on
/dev/vda13      961M  657M  305M  69% /local
```

With `--boot-image-size 20G`, the boot disk image is resized to 20 GiB before boot, and Bottlerocket extends the data partition accordingly:

```
bash-5.1# df -h /local
Filesystem      Size  Used Avail Use% Mounted on
/dev/vda13       18G  781M   18G   5% /local
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

markusboehme committed:
start-local-vm: update usage text to match current parameters

Add new options and metasyntactic variables to the usage section.
Perform some word smithing on the --product-name option while there.

Signed-off-by: Markus Boehme <markubo@amazon.com>

markusboehme committed:
start-local-vm: allow resizing the extracted images before boot

Sometimes it is helpful to not go with the default image sizes for
testing purposes--in practice, this would mean testing with bigger
images. Introduce the `--os-image-size` and `--data-image-size`
options to `start-local-vm` that allow for resizing the respective
images after they have been extracted, but before they are booted.

Signed-off-by: Markus Boehme <markubo@amazon.com>

bcressey commented on tools/start-local-vm:
```diff
@@ -80,6 +83,8 @@ Options:
                         private partition will be lost
     --firmware-code     override the default firmware executable file
     --firmware-vars     override the initial firmware variable storage file
+    --boot-image-size   resize the boot disk image to the given size (e.g. 4096M)
```
nit: in `buildsys` we refer to this as the [os image](https://github.com/bottlerocket-os/bottlerocket/blob/baef0849029ba7708033f8b350ef91780d3eff91/tools/buildsys/src/manifest.rs#L144):
```suggestion
    --os-image-size   resize the OS disk image to the given size (e.g. 4096M)
```

markusboehme replied:
Good point! Makes a lot of sense to stay consistent here. I added a commit that adopts the existing terminology for the `boot_image` variable names while I'm at it.

bcressey commented on tools/start-local-vm:
```diff
@@ -80,6 +83,8 @@ Options:
     --firmware-code     override the default firmware executable file
     --firmware-vars     override the initial firmware variable storage file
+    --boot-image-size   resize the boot disk image to the given size (e.g. 4096M)
+    --data-image-size   reisze the data disk image to the given size (e.g. 20G)
```
```suggestion
    --data-image-size   resize the data disk image to the given size (e.g. 20G)
```

markusboehme replied:
Ran a spell checker on the script and fixed another typo

markusboehme committed:
start-local-vm: match variable names to buildsys terminology

As Ben Cressey rightfully pointed out, the `start-local-vm` script is
using slightly different terminology than buildsys when referring to the
final Bottlerocket images. Adopt the existing terminology to avoid
confusion by renaming "boot images" to "OS images".

Signed-off-by: Markus Boehme <markubo@amazon.com>

foersleo reviewed (approved):
Great improvement. Also happy that you took the time to do some code hygiene.
